File: tests/integration/bec_transport.py
#!/usr/bin/env python
from xpdeint.XSILFile import XSILFile

xsilFile = XSILFile("bec_transport.xsil")

def firstElementOrNone(enumerable):
  for element in enumerable:
    return element
  return None

t_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].independentVariables if _["name"] == "t")
x_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].independentVariables if _["name"] == "x")
mean_psi_real_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].dependentVariables if _["name"] == "mean_psi_real")
mean_psi_imag_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].dependentVariables if _["name"] == "mean_psi_imag")
mean_density_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].dependentVariables if _["name"] == "mean_density")
stderr_psi_real_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].dependentVariables if _["name"] == "stderr_psi_real")
stderr_psi_imag_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].dependentVariables if _["name"] == "stderr_psi_imag")
stderr_density_1 = firstElementOrNone(_["array"] for _ in xsilFile.xsilObjects[0].dependentVariables if _["name"] == "stderr_density")

# Write your plotting commands here.
# You may want to import pylab (from pylab import *) or matplotlib (from matplotlib import *)

print(x_1)

File: tests/integration/bec_transport.m
if (exist('OCTAVE_VERSION', 'builtin')) % Octave
  load bec_transport.h5
  t_1 = eval('_1.t');
  x_1 = eval('_1.x');
  mean_psi_real_1 = eval('_1.mean_psi_real');
  mean_psi_real_1 = permute(mean_psi_real_1, ndims(mean_psi_real_1):-1:1);
  mean_psi_imag_1 = eval('_1.mean_psi_imag');
  mean_psi_imag_1 = permute(mean_psi_imag_1, ndims(mean_psi_imag_1):-1:1);
  mean_density_1 = eval('_1.mean_density');
  mean_density_1 = permute(mean_density_1, ndims(mean_density_1):-1:1);
  stderr_psi_real_1 = eval('_1.stderr_psi_real');
  stderr_psi_real_1 = permute(stderr_psi_real_1, ndims(stderr_psi_real_1):-1:1);
  stderr_psi_imag_1 = eval('_1.stderr_psi_imag');
  stderr_psi_imag_1 = permute(stderr_psi_imag_1, ndims(stderr_psi_imag_1):-1:1);
  stderr_density_1 = eval('_1.stderr_density');
  stderr_density_1 = permute(stderr_density_1, ndims(stderr_density_1):-1:1);
  clear _1;
else % MATLAB
  t_1 = hdf5read('bec_transport.h5', '/1/t');
  x_1 = hdf5read('bec_transport.h5', '/1/x');
  mean_psi_real_1 = hdf5read('bec_transport.h5', '/1/mean_psi_real');
  mean_psi_real_1 = permute(mean_psi_real_1, ndims(mean_psi_real_1):-1:1);
  mean_psi_imag_1 = hdf5read('bec_transport.h5', '/1/mean_psi_imag');
  mean_psi_imag_1 = permute(mean_psi_imag_1, ndims(mean_psi_imag_1):-1:1);
  mean_density_1 = hdf5read('bec_transport.h5', '/1/mean_density');
  mean_density_1 = permute(mean_density_1, ndims(mean_density_1):-1:1);
  stderr_psi_real_1 = hdf5read('bec_transport.h5', '/1/stderr_psi_real');
  stderr_psi_real_1 = permute(stderr_psi_real_1, ndims(stderr_psi_real_1):-1:1);
  stderr_psi_imag_1 = hdf5read('bec_transport.h5', '/1/stderr_psi_imag');
  stderr_psi_imag_1 = permute(stderr_psi_imag_1, ndims(stderr_psi_imag_1):-1:1);
  stderr_density_1 = hdf5read('bec_transport.h5', '/1/stderr_density');
  stderr_density_1 = permute(stderr_density_1, ndims(stderr_density_1):-1:1);
end
